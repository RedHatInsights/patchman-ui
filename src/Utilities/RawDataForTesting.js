// raw data for testing purposes only

export const advisoryRows = [{
    attributes: {
        advisory_type: 1,
        applicable_systems: 1,
        description: 'The microcode_ctl ',
        public_date: '2020-06-24T17:22:25Z',
        synopsis: 'microcode_ctl',
        severity: 3
    },
    id: 'RHEA-2020:2743',
    type: 'advisory'
}];

export const systemAdvisoryRows = [{
    attributes: {
        advisory_type: 3,
        description: 'Kernel-based Virtual.',
        public_date: '2020-06-30T13:38:53Z',
        severity: 3,
        synopsis: 'Important'
    },
    id: 'RHSA-2020:2774',
    type: 'advisory'
}];

export const systemRows = [{
    attributes: {
        display_name: 'automation_host',
        last_evaluation: '2020-08-12T13:57:54.028883Z',
        last_upload: '2020-08-12T09:26:33.891907Z',
        packages_installed: 0,
        packages_updatable: 0,
        rhba_count: 213,
        rhea_count: 35,
        rhsa_count: 77,
        stale: false
    },
    id: '8ddb54f5-aeeb-49b1-8448-c29049d686c1',
    type: 'system'
}];

export const advisoryHeader = {
    attributes: {
        advisory_type: 3,
        description: 'Kernel-based Virtual.',
        public_date: '2020-06-30T13:38:53Z',
        modified_date: '2020-06-30T13:38:53Z',
        severity: 3,
        synopsis: 'Important'
    },
    id: 'RHSA-2020:2774',
    type: 'advisory'
};

export const advisoryDetailRows = {
    data: {
        attributes: {
            cves: ['CVE-2020-11080'],
            description: 'libnghttp2.',
            fixes: null,
            modified_date: '2020-06-25T16:31:29Z',
            packages: { libnghttp2: '1.33.0-3.el8_2.1.x86_64' },
            public_date: '2020-06-25T16:31:29Z',
            references: [],
            severity: 3,
            solution: 'For details',
            synopsis: 'Important: ',
            topic: 'An update .'
        }
    },
    id: 'RHSA-2020:2755',
    type: 'advisory'
};
export const systemPackages = [{
    description: 'This package contains testing data',
    evra: '2.2.*',
    name: 'acl',
    summary: 'Access control list utilities',
    updates: null
}];

export const entityDetail = {
    loaded: true,
    activeApps: [],
    entity: {
        account: '60'
    }
};
