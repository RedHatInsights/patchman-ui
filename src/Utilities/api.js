import { encodeApiParams } from './Helpers';

export function createApiCall(
    endpoint,
    method,
    parameters = undefined,
    data = undefined
) {
    if (parameters && method === 'get') {
        endpoint = endpoint.concat(encodeApiParams(parameters));
    }

    let result = window.insights.chrome.auth
    .getUser()
    .then(() =>
        fetch('/api/patch/v1' + endpoint, {
            method,
            credentials: 'include',
            body: JSON.stringify(data)
        })
    )
    .then(res => {
        if (!res.ok) {
            const contentType = res.headers.get('content-type');
            if (contentType.indexOf('json') !== -1) {
                throw res;
            } else {
                throw {
                    errors: [
                        { status: res.status, detail: res.statusText }
                    ]
                };
            }
        }

        return res.json();
    })
    .catch(caughtError => {
        const error = Promise.resolve(caughtError || {});
        const genericError = {
            title:
                    'There was an error getting data'
        };
        return error.then(error => {
            const [info] = error.errors;
            const result = error.errors && { ...genericError, detail: info.detail, status: info.status } || genericError;
            throw result ;
        });
    });
    return result;
}

export const fetchApplicableAdvisoriesApi = params => {
    return createApiCall('/advisories', 'get', params);
};

export const fetchApplicableSystemAdvisoriesApi = params => {
    let { id, ...allParams } = params;
    return createApiCall(`/systems/${id}/advisories`, 'get', allParams);
};

export const fetchSystems = params => {
    return createApiCall('/systems', 'get', params);
};

export const fetchAdvisoryDetailsApi = params => {
    return createApiCall(`/advisories/${params.advisoryName}`, 'get');
};

export const fetchApplicablePackagesApi = params => {
    let { id, ...allParams } = params;
    return createApiCall(`/systems/${id}/packages`, 'get', allParams);
};

export const fetchAffectedSystems = params => {
    const { id, ...args } = params;
    return createApiCall(`/advisories/${id}/systems`, 'get', args);
};

export const fetchPackagesList = params => {
    return createApiCall('/packages', 'get', params);
};

export const exportAdvisoriesCSV = params => {
    let endpoint = '/export/advisories';
    endpoint = endpoint.concat(encodeApiParams(params));
    return fetch('/api/patch/v1' + endpoint, {
        method: 'get',
        credentials: 'include',
        headers: new Headers({ accept: 'text/csv' })
    }).then(res => res.text());
};

export const exportAdvisoriesJSON = params => {
    let endpoint = '/export/advisories';
    endpoint = endpoint.concat(encodeApiParams(params));
    return fetch('/api/patch/v1' + endpoint, {
        method: 'get',
        credentials: 'include',
        headers: new Headers({ accept: 'application/json' })
    }).then(res => res.json());
};

export const exportSystemsCSV = params => {
    let endpoint = '/export/systems';
    endpoint = endpoint.concat(encodeApiParams(params));
    return fetch('/api/patch/v1' + endpoint, {
        method: 'get',
        credentials: 'include',
        headers: new Headers({ accept: 'text/csv' })
    }).then(res => res.text());
};

export const exportSystemsJSON = params => {
    let endpoint = '/export/systems';
    endpoint = endpoint.concat(encodeApiParams(params));
    return fetch('/api/patch/v1' + endpoint, {
        method: 'get',
        credentials: 'include',
        headers: new Headers({ accept: 'application/json' })
    }).then(res => res.json());
};
