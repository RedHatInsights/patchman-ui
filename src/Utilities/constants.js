import BugIcon from '@patternfly/react-icons/dist/js/icons/bug-icon';
import EnhancementIcon from '@patternfly/react-icons/dist/js/icons/enhancement-icon';
import SecurityIcon from '@patternfly/react-icons/dist/js/icons/security-icon';
import React from 'react';
import { subtractDate } from './Helpers';

export const STATUS_REJECTED = 'rejected';
export const STATUS_LOADING = 'loading';
export const STATUS_RESOLVED = 'resolved';

export const compoundSortValues = {
    operating_system: {
        asc: 'os_name,os_major,os_minor',
        desc: '-os_name,-os_major,-os_minor'
    },
    applicable_advisories: {
        asc: 'rhsa_count,rhba_count,rhea_count',
        desc: '-rhsa_count,-rhba_count,-rhea_count'
    }
};

// messy check because of frontend-components tests
export const ENABLE_PACKAGES = window?.insights?.chrome?.isBeta && insights?.chrome?.isBeta();
export const storeListDefaults = {
    rows: [],
    metadata: {
        limit: 25,
        offset: 0,
        total_items: 0
    },
    expandedRows: {},
    selectedRows: {},
    queryParams: {},
    error: {}
};

export const publicDateOptions = [
    { apiValue: '', label: 'All', value: 'all' },
    {
        apiValue: `gt:${subtractDate(7)}`,
        label: 'Last 7 days',
        value: 'last7'
    },
    {
        apiValue: `gt:${subtractDate(30)}`,
        label: 'Last 30 days',
        value: 'last30'
    },
    {
        apiValue: `gt:${subtractDate(90)}`,
        label: 'Last 90 days',
        value: 'last90'
    },
    {
        apiValue: `gt:${subtractDate(365)}`,
        label: 'Last year',
        value: 'lastYear'
    },
    {
        apiValue: `lt:${subtractDate(365)}`,
        label: 'More than 1 year ago',
        value: 'moreThanYear'
    }
];

export const advisorySeverities = [
    {
        value: 0,
        label: 'N/A',
        color: 'var(--pf-global--Color--200)'
    },
    {
        value: 1,
        label: 'Low',
        color: 'var(--pf-global--Color--200)',
        text:
            'This rating is given to all other issues that have a security impact. These are the types of vulnerabilities' +
            ' that are believed to require unlikely circumstances to be able to be exploited, or where a successful exploit' +
            ' would give minimal consequences.'
    },
    {
        value: 2,
        label: 'Moderate',
        color: 'var(--pf-global--warning-color--100)',
        text:
            'This rating is given to flaws that may be more difficult to exploit but could still lead to some' +
            ' compromise of the confidentiality, integrity, or availability of resources, under certain circumstances.' +
            ' These are the types of vulnerabilities that could have had a Critical impact or Important impact' +
            ' but are less easily exploited based on a technical evaluation of the flaw, or affect unlikely' +
            ' configurations.'
    },
    {
        value: 3,
        label: 'Important',
        color: '#ec7a08',
        text:
            'This rating is given to flaws that can easily compromise the confidentiality, integrity, or availability' +
            ' of resources. These are the types of vulnerabilities that allow local users to gain privileges, allow' +
            ' unauthenticated remote users to view resources that should otherwise be protected by authentication,' +
            ' allow authenticated remote users to execute arbitrary code, or allow remote users to cause a denial' +
            ' of service.'
    },
    {
        value: 4,
        label: 'Critical',
        color: 'var(--pf-global--danger-color--100)',
        text:
            'This rating is given to flaws that could be easily exploited by a remote unauthenticated attacker' +
            ' and lead to system compromise (arbitrary code execution) without requiring user interaction.' +
            ' These are the types of vulnerabilities that can be exploited by worms. Flaws that require an authenticated' +
            ' remote user, a local user, or an unlikely configuration are not classed as Critical impact.'
    }
];

export const advisoryTypes = [
    {
        value: 3,
        label: 'Security',
        icon: <SecurityIcon />
    },
    {
        value: 2,
        label: 'Bugfix',
        icon: <BugIcon />
    },
    {
        value: 1,
        label: 'Enhancement',
        icon: <EnhancementIcon />
    }
];

export const updatableTypes = [
    {
        value: '',
        label: 'All'
    },
    {
        value: true,
        label: 'Upgradable'
    }
];

export const packagesListUpdatableTypes = [
    {
        value: '',
        label: 'All'
    },
    {
        value: 'gt:0',
        label: 'Upgradable'
    }
];

export const filterCategories = {
    advisory_type: {
        label: 'Advisory type',
        values: advisoryTypes
    },
    public_date: {
        label: 'Public date',
        values: publicDateOptions
    },
    updatable: {
        label: 'Status',
        values: updatableTypes
    },
    systems_updatable: {
        label: 'Status',
        values: packagesListUpdatableTypes
    }
};

export const entityTypes = {
    advisories: 'advisories',
    packages: 'packages'
};

export const ReadOnlyNotification = {
    title: 'title',
    detail: 'message'
};

export const remediationIdentifiers = {
    package: 'patch-package',
    advisory: 'patch-advisory'
};
