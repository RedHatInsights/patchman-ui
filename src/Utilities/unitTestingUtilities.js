import { useSelector } from 'react-redux';

/* eslint-disable */
export const initMocks = () => {

    window.insights = {
        loadInventory: (args) => new Promise((resolve, reject) =>  {
            resolve(({
                inventoryConnector: (args) => {
                            const InventoryTable = ({ children }) => <div >A mock passed! {children} </div>;
                            return ({ InventoryTable });
                        },
                        mergeWithEntities: () => {}
            }));
        }),
        loadRemediations: () => new Promise((resolve, reject) =>  {
            resolve({ 
                openWizard: () => new Promise((resolve, reject) =>  { 
                    resolve(true)
                })
            });
        })
    };

};

export const mockStore = (initialState, mutatedState) => {
    const customMiddleWare = store => next => action => {
        useSelector.mockImplementation(callback => {
            return callback(mutatedState);
        });
        next(action);
    };

    const mockStore = configureStore([customMiddleWare]);

    return mockStore(initialState);
}
/* eslint-enable */
