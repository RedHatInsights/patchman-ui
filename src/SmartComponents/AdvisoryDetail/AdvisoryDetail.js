import { Stack, StackItem, Text, TextContent, TextVariants } from '@patternfly/react-core';
import { Main } from '@redhat-cloud-services/frontend-components';
import propTypes from 'prop-types';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { withRouter } from 'react-router-dom';
import messages from '../../Messages';
import AdvisoryHeader from '../../PresentationalComponents/AdvisoryHeader/AdvisoryHeader';
import Header from '../../PresentationalComponents/Header/Header';
import { Unavailable } from '@redhat-cloud-services/frontend-components';
import { paths } from '../../Routes';
import { clearAdvisoryDetailStore, clearAdvisorySystemsStore, fetchAvisoryDetails } from '../../store/Actions/Actions';
import { STATUS_LOADING, STATUS_REJECTED } from '../../Utilities/constants';
import { setPageTitle } from '../../Utilities/Hooks';
import { intl } from '../../Utilities/IntlProvider';
import AdvisorySystems from '../AdvisorySystems/AdvisorySystems';

const AdvisoryDetail = ({ match }) => {
    const dispatch = useDispatch();
    const [advisoryName] = React.useState(match.params.advisoryId);

    const pageTitle = `${advisoryName} - ${intl.formatMessage(messages.titlesAdvisories)}`;
    setPageTitle(pageTitle);

    const advisoryDetails = useSelector(
        ({ AdvisoryDetailStore }) => AdvisoryDetailStore
    );
    const status = useSelector(
        ({ AdvisoryDetailStore }) => AdvisoryDetailStore.status
    );

    React.useEffect(() => {
        dispatch(fetchAvisoryDetails({ advisoryName }));
    }, []);

    React.useEffect(() => {
        return () => {
            dispatch(clearAdvisorySystemsStore());
            dispatch(clearAdvisoryDetailStore());
        };
    }, []);

    const { attributes } = advisoryDetails.data;
    return (
        <React.Fragment>
            <Header
                title={advisoryName}
                headerOUIA={'advisory-details'}
                breadcrumbs={[
                    {
                        title: intl.formatMessage(messages.titlesPatchAdvisories),
                        to: paths.advisories.to,
                        isActive: false
                    },
                    {
                        title: advisoryName,
                        isActive: true
                    }
                ]}
            >{status === STATUS_REJECTED ? <Unavailable /> :
                    <AdvisoryHeader
                        attributes={{ ...attributes, id: advisoryName }}
                        isLoading={status === STATUS_LOADING}
                    />}
            </Header>
            <Main>
                <Stack hasGutter>
                    <StackItem>
                        <TextContent>
                            <Text component={TextVariants.h2}>{intl.formatMessage(messages.titlesAffectedSystems)}</Text>
                        </TextContent>
                    </StackItem>
                    <StackItem>
                        <AdvisorySystems advisoryName={advisoryName} />
                    </StackItem>
                </Stack>
            </Main>
        </React.Fragment>
    );
};

AdvisoryDetail.propTypes = {
    match: propTypes.any
};

export default withRouter(AdvisoryDetail);
