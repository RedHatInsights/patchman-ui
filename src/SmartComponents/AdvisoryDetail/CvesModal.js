import React, { useState, useMemo } from 'react';
import { Modal } from '@patternfly/react-core';
import messages from '../../Messages';
import { intl } from '../../Utilities/IntlProvider';
import TableView from '../../PresentationalComponents/TableView/TableView';
import searchFilter from '../../PresentationalComponents/Filters/SearchFilter';
import { cvesTableColumns } from '../../PresentationalComponents/TableView/TableViewAssets';
import { useDispatch, useSelector } from 'react-redux';
import { fetchCves } from '../../store/Actions/Actions';
import propTypes from 'prop-types';
import { createCvesRows } from '../../Utilities/DataMappers';
import { sortCves } from '..//../Utilities/Helpers';
import { Unavailable } from '@redhat-cloud-services/frontend-components';
import { STATUS_REJECTED } from '../../Utilities/constants';
import { SortByDirection } from '@patternfly/react-table';

const CvesModal = ({ cveIds }) =>{
    const dispatch = useDispatch();
    const [cves, setCves] = useState([]);
    const [rows, setRows] = useState([]);
    const [page, setPage] = useState(1);
    const [perPage, setPerPage] = useState(10);
    const [search, setSearch] = useState(undefined);
    const [sortBy, setSortBy] = useState({
        direction: SortByDirection.asc,
        index: 0
    });

    const data = useSelector(({ CvesListStore }) => CvesListStore.rows);

    const status = useSelector(
        ({ CvesListStore }) => CvesListStore.status
    );

    React.useEffect(() => {
        dispatch(fetchCves({ cveIds }));
    }, []);

    React.useMemo(() => {
        setRows(cves.slice((page - 1) * perPage, page * perPage));
    }, [cves, page, perPage, sortBy]);

    useMemo(() => {
        const sortedCves = (search !== undefined && search !== '')
            && data.filter(
                cve => {
                    const { attributes: { synopsis } } = cve;
                    return synopsis.toLowerCase().includes(search);
                }
            ) || data;

        setCves(createCvesRows((sortedCves.length !== 0 || search) && sortedCves || data));
    }, [search, data]);

    const handleClose = () => {
        setRows(undefined);
    };

    const handleFilter = ({ search }) =>{
        setPage(page);
        setSearch(search);
    };

    const handlePageChange = (_, page) => {
        setPage(page);
    };

    const handlePerPageChange = (_, perPage) => {
        setPage(1);
        setPerPage(perPage);
    };

    const handleSort = (_, index, direction) => {
        const { sortBy, sortedCves } = sortCves(cves, index, direction);

        setSortBy(sortBy);
        setCves(sortedCves);
    };

    const errorState = status === STATUS_REJECTED && <Unavailable />;

    return (
        <React.Fragment>
            <Modal
                variant='small'
                title={intl.formatMessage(messages.labelsCves)}
                isOpen={Boolean(rows)}
                onClose={handleClose}
            >
                <TableView
                    columns={cvesTableColumns}
                    onSetPage={handlePageChange}
                    onPerPageSelect={handlePerPageChange}
                    apply={handleFilter}
                    tableOUIA={'cves-table'}
                    paginationOUIA={'cves-pagination'}
                    onSort={handleSort}
                    sortBy={sortBy}
                    store={{
                        rows,
                        metadata: { limit: perPage, offset: (page - 1) * perPage, total_items: cves && cves.length },
                        status, queryParams: { filter: {}, search }
                    }}
                    filterConfig={{
                        items: [
                            searchFilter(handleFilter, search, intl.formatMessage(messages.labelsFiltersCvesSearch))
                        ]
                    }}
                    errorState={errorState}
                />
            </Modal>
        </React.Fragment>
    );

};

CvesModal.propTypes = {
    cveIds: propTypes.array
};

export default CvesModal;
