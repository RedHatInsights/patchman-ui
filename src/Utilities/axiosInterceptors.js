import {
    authInterceptor,
    responseDataInterceptor
} from '@redhat-cloud-services/frontend-components-utilities/files/interceptors';
import axios from 'axios';
import { ReadOnlyNotification } from './constants';
const axiosInstance = axios.create();

export async function  accessCheckInterceptor (config)  {

    const userPermissions = await insights.chrome.getUserPermissions();
    const inventoryPermissionList = ['inventory:*:*', 'inventory:*:read', 'inventory:hosts:read'];

    const hasInventoryAccess = userPermissions.some((access) => inventoryPermissionList.includes(access?.permission));

    if (!hasInventoryAccess) {
        throw new axios.Cancel('Operation canceled. User does not have access to Inventory');
    }

    return config;
}

export function errorInterceptor(err) {

    if (!axios.isCancel(err)) {

        const { response, isAxiosError } = { ...err };

        if (response && isAxiosError) {
            const { status, statusText, data } = response;

            if (!status) {
                throw err;
            } else {
                const genericError = {
                    title:
                            'There was an error getting data'
                };

                const result = { ...genericError, detail: data.error && data.error || statusText, status };

                throw result;
            }
        }

        throw err;
    }
}

export function readOnlyInterceptor(error) {
    if (error.response && error.response.status === 503) {
        const data = ReadOnlyNotification;
        throw data;
    }

    throw error;
}

axiosInstance.interceptors.request.use(accessCheckInterceptor);
axiosInstance.interceptors.request.use(authInterceptor);
axiosInstance.interceptors.response.use(responseDataInterceptor);
axiosInstance.interceptors.response.use(null, readOnlyInterceptor);
axiosInstance.interceptors.response.use(null, errorInterceptor);

export default axiosInstance;
