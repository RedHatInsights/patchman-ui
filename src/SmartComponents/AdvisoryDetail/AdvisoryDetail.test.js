import toJson from 'enzyme-to-json';
import { act } from 'react-dom/test-utils';
import { Provider, useSelector } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { storeListDefaults } from '../../Utilities/constants';
import { advisoryDetailRows } from '../../Utilities/RawDataForTesting';
import { initMocks } from '../../Utilities/unitTestingUtilities.js';
import AdvisoryDetail from './AdvisoryDetail';

initMocks();

jest.mock('../../PresentationalComponents/AdvisoryHeader/AdvisoryHeader', () =>
    () => <div> hello </div>
);

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn()
}));

jest.mock('../AdvisorySystems/AdvisorySystems', () =>
    () =>  <div> hello </div>
);

const mockState = { ...storeListDefaults, ...advisoryDetailRows };

const initStore = (state) => {
    const customMiddleWare = () => next => action => {
        useSelector.mockImplementation(callback => {
            return callback({  AdvisoryDetailStore: state });
        });
        next(action);
    };

    const mockStore = configureStore([customMiddleWare]);
    return mockStore({  AdvisoryDetailStore: state });
};

let wrapper;
let store = initStore(mockState);

beforeEach(() => {
    store.clearActions();
    useSelector.mockImplementation(callback => {
        return callback({ AdvisoryDetailStore: mockState });
    });
    wrapper = mount(<Provider store={store}>
        <Router><AdvisoryDetail /></Router>
    </Provider>);
});

afterEach(() => {
    useSelector.mockClear();
});

describe('AdvisoryDetail.js', () => {
    it('Should match the snapshots', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should clear store on unmount', async () => {
        let wrapper;
        await act(async() => {
            wrapper = mount(
                <Provider store={store}>
                    <Router><AdvisoryDetail /></Router>
                </Provider>
            );
        });
        act(() => {
            wrapper.unmount();
        });
        const dispatchedActions = store.getActions();
        expect(dispatchedActions.filter(item => item.type === 'CLEAR_AFFECTED_SYSTEMS')).toHaveLength(1);
        expect(dispatchedActions.filter(item => item.type === 'CLEAR_ADVISORY_DETAILS')).toHaveLength(1);
    });

    it('Should display error page when status is rejected', () => {

        const rejectedState = { ...mockState, status: 'rejected', error: { detail: 'test' } };

        useSelector.mockImplementation(callback => {
            return callback({ AdvisoryDetailStore: rejectedState });
        });

        const tempStore = initStore(rejectedState);
        const tempWrapper = mount(<Provider store={tempStore}>
            <Router><AdvisoryDetail/></Router>
        </Provider>);
        expect(tempWrapper.find('Error')).toBeTruthy();
    });

});
