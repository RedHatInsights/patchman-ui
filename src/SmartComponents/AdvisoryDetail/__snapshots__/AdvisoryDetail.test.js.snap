// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvisoryDetail.js Should match the snapshots 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(AdvisoryDetail)>
        <AdvisoryDetail
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <Header
            breadcrumbs={
              Array [
                Object {
                  "isActive": false,
                  "title": "Patch",
                  "to": "/",
                },
                Object {
                  "isActive": false,
                  "title": "Advisories",
                  "to": "/",
                },
                Object {
                  "isActive": true,
                  "title": undefined,
                },
              ]
            }
            headerOUIA="advisory-details"
          >
            <PageHeader
              data-ouia-component-type="advisory-details-page-header"
            >
              <section
                className="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
                data-ouia-component-type="advisory-details-page-header"
                widget-type="InsightsPageHeader"
              >
                <HeaderBreadcrumbs
                  headerOUIA="advisory-details"
                  items={
                    Array [
                      Object {
                        "isActive": false,
                        "title": "Patch",
                        "to": "/",
                      },
                      Object {
                        "isActive": false,
                        "title": "Advisories",
                        "to": "/",
                      },
                      Object {
                        "isActive": true,
                        "title": undefined,
                      },
                    ]
                  }
                >
                  <Breadcrumb>
                    <nav
                      aria-label="Breadcrumb"
                      className="pf-c-breadcrumb"
                      data-ouia-component-id={0}
                      data-ouia-component-type="PF4/Breadcrumb"
                      data-ouia-safe={true}
                    >
                      <ol
                        className="pf-c-breadcrumb__list"
                      >
                        <BreadcrumbItem
                          isActive={false}
                          key=".$Patch"
                          showDivider={false}
                        >
                          <li
                            className="pf-c-breadcrumb__item"
                          >
                            <Link
                              data-ouia-component-id="breadcrumb-to-Patch"
                              data-ouia-component-type="advisory-details-breadcrumb"
                              to="/"
                            >
                              <LinkAnchor
                                data-ouia-component-id="breadcrumb-to-Patch"
                                data-ouia-component-type="advisory-details-breadcrumb"
                                href="/"
                                navigate={[Function]}
                              >
                                <a
                                  data-ouia-component-id="breadcrumb-to-Patch"
                                  data-ouia-component-type="advisory-details-breadcrumb"
                                  href="/"
                                  onClick={[Function]}
                                >
                                  Patch
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                        </BreadcrumbItem>
                        <BreadcrumbItem
                          isActive={false}
                          key=".$Advisories"
                          showDivider={true}
                        >
                          <li
                            className="pf-c-breadcrumb__item"
                          >
                            <span
                              className="pf-c-breadcrumb__item-divider"
                            >
                              <AngleRightIcon
                                color="currentColor"
                                noVerticalAlign={false}
                                size="sm"
                              >
                                <svg
                                  aria-hidden={true}
                                  aria-labelledby={null}
                                  fill="currentColor"
                                  height="1em"
                                  role="img"
                                  style={
                                    Object {
                                      "verticalAlign": "-0.125em",
                                    }
                                  }
                                  viewBox="0 0 256 512"
                                  width="1em"
                                >
                                  <path
                                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                                    transform=""
                                  />
                                </svg>
                              </AngleRightIcon>
                            </span>
                            <Link
                              data-ouia-component-id="breadcrumb-to-Advisories"
                              data-ouia-component-type="advisory-details-breadcrumb"
                              to="/"
                            >
                              <LinkAnchor
                                data-ouia-component-id="breadcrumb-to-Advisories"
                                data-ouia-component-type="advisory-details-breadcrumb"
                                href="/"
                                navigate={[Function]}
                              >
                                <a
                                  data-ouia-component-id="breadcrumb-to-Advisories"
                                  data-ouia-component-type="advisory-details-breadcrumb"
                                  href="/"
                                  onClick={[Function]}
                                >
                                  Advisories
                                </a>
                              </LinkAnchor>
                            </Link>
                          </li>
                        </BreadcrumbItem>
                        <BreadcrumbItem
                          isActive={true}
                          key=".2"
                          showDivider={true}
                        >
                          <li
                            className="pf-c-breadcrumb__item"
                          >
                            <span
                              className="pf-c-breadcrumb__item-divider"
                            >
                              <AngleRightIcon
                                color="currentColor"
                                noVerticalAlign={false}
                                size="sm"
                              >
                                <svg
                                  aria-hidden={true}
                                  aria-labelledby={null}
                                  fill="currentColor"
                                  height="1em"
                                  role="img"
                                  style={
                                    Object {
                                      "verticalAlign": "-0.125em",
                                    }
                                  }
                                  viewBox="0 0 256 512"
                                  width="1em"
                                >
                                  <path
                                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                                    transform=""
                                  />
                                </svg>
                              </AngleRightIcon>
                            </span>
                          </li>
                        </BreadcrumbItem>
                      </ol>
                    </nav>
                  </Breadcrumb>
                </HeaderBreadcrumbs>
                <PageHeaderTitle>
                  <Title
                    className=""
                    headingLevel="h1"
                    size="2xl"
                    widget-type="InsightsPageHeaderTitle"
                  >
                    <h1
                      className="pf-c-title pf-m-2xl"
                      widget-type="InsightsPageHeaderTitle"
                    >
                       
                       
                    </h1>
                  </Title>
                </PageHeaderTitle>
                <Component
                  attributes={
                    Object {
                      "cves": Array [
                        "CVE-2020-11080",
                      ],
                      "description": "libnghttp2.",
                      "fixes": null,
                      "id": undefined,
                      "modified_date": "2020-06-25T16:31:29Z",
                      "packages": Object {
                        "libnghttp2": "1.33.0-3.el8_2.1.x86_64",
                      },
                      "public_date": "2020-06-25T16:31:29Z",
                      "references": Array [],
                      "severity": 3,
                      "solution": "For details",
                      "synopsis": "Important: ",
                      "topic": "An update .",
                    }
                  }
                  isLoading={false}
                >
                  <div>
                     hello 
                  </div>
                </Component>
              </section>
            </PageHeader>
          </Header>
          <Connect(Main)>
            <Main>
              <section
                className="pf-l-page__main-section pf-c-page__main-section"
                page-type=""
              >
                <Stack
                  hasGutter={true}
                >
                  <div
                    className="pf-l-stack pf-m-gutter"
                  >
                    <StackItem>
                      <div
                        className="pf-l-stack__item"
                      >
                        <TextContent>
                          <div
                            className="pf-c-content"
                          >
                            <Text
                              component="h2"
                            >
                              <h2
                                className=""
                                data-pf-content={true}
                              >
                                Affected systems
                              </h2>
                            </Text>
                          </div>
                        </TextContent>
                      </div>
                    </StackItem>
                    <StackItem>
                      <div
                        className="pf-l-stack__item"
                      >
                        <Component>
                          <div>
                             hello 
                          </div>
                        </Component>
                      </div>
                    </StackItem>
                  </div>
                </Stack>
              </section>
            </Main>
          </Connect(Main)>
        </AdvisoryDetail>
      </withRouter(AdvisoryDetail)>
    </Router>
  </BrowserRouter>
</Provider>
`;
